dlrover/python/diagnosis/common/diagnosis_action.py

```diff
@@ -274,6 +275,31 @@ def __repr__(self):
         )
 
 
+class ObservationAction(DiagnosisAction):
```

Comment:
Why define 'observation' as a 'action'? 
My understanding is that observation, as an abstraction, takes metrics as input and outputs actions.

Reply:
It's a trade off here. If programmers want to generate the action in `observe` here, they need not define the `resolve`.